### PR TITLE
fix: prevent null token

### PR DESCRIPTION
**Pull request by skjnldsv**
I have no clue how that can happen, because we clearly have the token in the URL, but :shrug: 

```json
{
  "reqId": "geRGVLc8I4Nvg8yDzXL6",
  "level": 3,
  "time": "2024-07-22T09:10:34+00:00",
  "remoteAddr": "213.163.39.210",
  "user": "--",
  "app": "index",
  "method": "GET",
  "url": "/s/hLOoTJBdpD78MAPL89N/download?path=OBFUSCATED.pdf.pdf",
  "message": "OCA\\Files_DownloadLimit\\Db\\LimitMapper::get(): Argument #1 ($token) must be of type string, null given, called in /var/www/html/apps/files_downloadlimit/lib/Listener/ShareLinkAccessedListener.php on line 59 in file '/var/www/html/apps/files_downloadlimit/lib/Db/LimitMapper.php' line 37",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0",
  "version": "29.0.3.4",
  "exception": {
    "Exception": "Exception",
    "Message": "OCA\\Files_DownloadLimit\\Db\\LimitMapper::get(): Argument #1 ($token) must be of type string, null given, called in /var/www/html/apps/files_downloadlimit/lib/Listener/ShareLinkAccessedListener.php on line 59 in file '/var/www/html/apps/files_downloadlimit/lib/Db/LimitMapper.php' line 37",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/html/lib/private/AppFramework/App.php",
        "line": 184,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          [
            "OCA\\Files_Sharing\\Controller\\ShareController"
          ],
          "downloadShare"
        ]
      },
```

### CODE REVIEW

File: lib/Listener/ShareLinkAccessedListener.php [hunk 1]
```diff
@@ -76,6 +76,8 @@ public function handle(Event $event): void {
 			$this->mapper->update($shareLimit);
 		} catch (DoesNotExistException $e) {
 			// No limit is set, ignore
+		} catch (\Exception $e) {
+			// Something went wrong, ignore
```
Comment (come-nc) [inline lib/Listener/ShareLinkAccessedListener.php:80]: Log it maybe if we do not know how to explain it? Or add an explicit check for missing token earlier and ignore it silently then. I’m always scared by catchall blocks with no log.

Comment (skjnldsv) [reply lib/Listener/ShareLinkAccessedListener.php]: Done

**Review by AndyScherzinger — APPROVED** [review]

**Comment by Antreesy** [comment]
Third time's the charm 😁
Thanks!

**Pull request merged**